c++/src/capnp/rpc.c++

```diff
@@ -858,6 +858,18 @@ private:
     }
   }
 
+  void waitAllAckedInBackground(kj::Own<RpcFlowController> flowController) {
+    // Calls `waitAllAcked()` on `flowController` as a background task. This is needed when an
+    // `RpcFlowController` is being abandoned but we don't actually intend to cancel the calls
+    // that were made on it.
+
+    // If we're disconnected, there's no need to wait for anything, so we'll just drop the
+    // flowController in that case.
+    if (connection.is<Connected>()) {
```

Comment:
Seems like the main change is the addition of this "connected" check, which makes sense given the description of the problem in the PR summary, although I must admit I don't yet have a firm enough grasp of the ownership graph to have tracked the problem down myself.

But, yeah, it looks like this check plus the `tasks.clear()` in `disconnect()` would maintain an invariant that ensures that `RpcConnectionState.tasks` will only contain flowController tasks for connections in the Connected state.